pandas/tests/test_window.py

```diff
@@ -289,6 +290,209 @@ def test_deprecations(self):
+                                  1: Series([self.naval, 3, 5, 7, 9])}),
+                'min': DataFrame({0: Series([self.naval, 0, 2, 4, 6]),
+                                  1: Series([self.naval, 1, 3, 5, 7])}),
+                'sum': DataFrame({0: Series([self.naval, 2, 6, 10, 14]),
+                                  1: Series([self.naval, 4, 8, 12, 16])}),
+                'mean': DataFrame({0: Series([self.naval, 1, 3, 5, 7]),
+                                  1: Series([self.naval, 2, 4, 6, 8])}),
+                'std': DataFrame({0: Series([self.naval] + [np.sqrt(2)] * 4),
+                                  1: Series([self.naval] + [np.sqrt(2)] * 4)}),
+                'var': DataFrame({0: Series([self.naval, 2, 2, 2, 2]),
+                                  1: Series([self.naval, 2, 2, 2, 2])}),
+                'median': DataFrame({0: Series([self.naval, 1, 3, 5, 7]),
+                                     1: Series([self.naval, 2, 4, 6, 8])}),
+            }
+        }
+        return expects
+
+    def _create_dtype_data(self, dtype):
+        sr1 = Series(range(5), dtype=dtype)
+        sr2 = Series(range(10, 0, -2), dtype=dtype)
+        df = DataFrame(np.arange(10).reshape((5, 2)), dtype=dtype)
+
+        data = {
+            'sr1': sr1,
+            'sr2': sr2,
+            'df': df
+        }
+
+        return data
+
+    def _create_data(self):
+        super(TestDtype, self)._create_data()
+        self.data = self._create_dtype_data(self.dtype)
+        self.expects = self.get_expects()
+
+    def setUp(self):
+        self._create_data()
+
+    def _cast_result(self, result, from_dtype, to_dtype):
+        if com.needs_i8_conversion(from_dtype):
+            if isinstance(result, Series):
+                result = result.view('i8')
+            elif isinstance(result, DataFrame):
+                final = []
+                for idx in result:
+                    final.append(Series(result[idx].view('i8')))
+                result = pd.concat(final, axis=1).reindex(
+                    columns=result.columns)
+        return result.astype(to_dtype)
+
+    def test_dtypes(self):
+        for f_name, d_name in product(self.funcs.keys(), self.data.keys()):
+            # Specify if the results and expectations
+            # need to be coerced to a given dtype
+            # once we changed the return value for roll_<function>,
+            # we should change coerce behavior here accordingly
+            f, res_dtype, exp_dtype = self.funcs[f_name]
+            d = self.data[d_name]
+            assert_equal = assert_series_equal if isinstance(
+                d, Series) else assert_frame_equal
+            exp = self.expects[d_name][f_name]
+            if exp_dtype:
+                exp = exp.astype(com.pandas_dtype(exp_dtype))
+
+            roll = d.rolling(window=self.window)
+            result = f(roll)
+            if res_dtype:
+                result = self._cast_result(result,
```

Comment:
pls take all casting out. The expected should already have the expectation of casts built in. If something is later changed, then it will obvious as the expecations are listed. They may be *wrong*, but that's ok for now (when I mean wrong they are not the 'correct' dtype as everything is getting casted ATM). This should be a dead simple tests. We don't want to keep expanding this, really make this much much simpler.